### PR TITLE
Factorize empty range error used in several parsers

### PR DESCRIPTION
**Pull request by Xaxetrov**

### CODE REVIEW

**Comment by Xaxetrov** [comment]
It is possible to also pass a variable like "xxx" or "xxx or yyy" and the EmptyRangeError would create "Expected ${variable} but got empty range. " instead of just "Empty range". Do you think that would be better ?

**Comment by Xaxetrov** [comment]
I can also implement it as a derived class from error with just a different constructor instead of a function. Do you think it is better ? Actually there are several more way to do this.

File: src/CommonErrors.cpp [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#include "CommonErrors.hpp"
+
+#include "Range.hpp"
+
+namespace cmake::language
+{
+  tl::unexpected<Error> EmptyRangeError(const Range& range)
```
Comment (FlorianChevassu) [inline src/CommonErrors.cpp]: I would name it `CreateEmptyRangeError`, `EmptyRangeError` sounds like a type name.

**Review by FlorianChevassu — COMMENTED** [review]
As you said, it would be nice to be able to pass an additional parameter to be able to customize the message. The "Expected ${variable}, but got empty range" sounds good to me.

**Comment by Xaxetrov** [comment]
> 
> 
> As you said, it would be nice to be able to pass an additional parameter to be able to customize the message. The "Expected ${variable}, but got empty range" sounds good to me.

Change done !

File: src/NewLineParser.cpp [hunk 2]
```diff
@@ -11,10 +12,7 @@ namespace cmake::language
   {
     if (r.IsEmpty())
     {
-      Error err;
-      err.message = "Empty range.";
-      err.context = r;
-      return tl::make_unexpected(err);
+      return CreateEmptyRangeError(r, R"(\n)");
```
Comment (FlorianChevassu) [inline src/NewLineParser.cpp]:
```suggestion
      return CreateEmptyRangeError(r, R"('\n')");
```

**Comment by FlorianChevassu** [comment]
Everything looks good to me ! Merging.

**Pull request merged**